### ISSUE TITLE
Remove the exclamation mark surrounded by a triangle when I add a new part

### ISSUE THREAD

**Issue by jamespblloyd** [SynBioDex/SBOLDesigner#258]
There is an exclamation mark surrounded by a triangle when I add a new part in v3.0. I have named the part but this remains. Can you please help me remove it. 
For an example image, see my tweet here: [https://twitter.com/JamesPBLloyd/status/1002391896478474241](url)

**Comment by cjmyers** [comment]
The exclamation mark is a warning indicator that there is no sequence for your part.  If you add a sequence, this warning should go away.

However, we really should have a tooltip to indicate what the warning means.

**Comment by cjmyers** [comment]
Also, change exclamation mark to show up when ANY component in the hierarchy is missing a sequence.

**Comment by cjmyers** [comment]
Tooltips:
Sequence incomplete <- exclamation
Composite <- if number of Components greater than zero
Combinatorial <- if the definition of the Component is referred to by a CombinatorialDerivation

**Comment by cjmyers** [comment]
Messages still need to match above.

**Comment by SamuelSauce** [comment]
Hey Chris. I was able to add the proper tooltips minus the Combinatorial. I commented out the code in my most recent push where I tried to account for Combinatorial, but it wouldn't work. What I tried was to grab the current SBOL document do a .getCombinatorialDerivations(), checking if the set contained the current component definition, but it is throwing an error when trying to get the SBOL document.

**Comment by michael13162** [comment]
You should get the CombinatorialDerivations, and for each CombinatorialDerivation, check if its template field points to the current CD.

**Comment by cjmyers** [comment]
Shouldn’t he be using the “hasVariants” method in SBOLDesign.java?

**Comment by michael13162** [comment]
Good point.  getToolTipText has a design element (e) passed in.  It seems like you can get all the information you want by calling:

e.getOrientation(), e.isComposite(), e.hasVariants(design, canvasCD), e.hasSequence()

**Comment by SamuelSauce** [comment]
So, I've finished the tooltip, with it properly displaying "Composite", "Incomplete" or "Combinatorial", but I still haven't finished recursing down the tree of sub-parts to figure out whether or not the sequence is incomplete for all sub-parts. Should I wait to create a pull request until this is done? If so, I've started with checking that the component is composite, using e.isComposite() and if true, then using e.component, but I'm stuck there as to figuring out how to retrieve the children of the specified component.

**Comment by cjmyers** [comment]
If you have the CD, you should be able to loop through the Components within the CD, and then explore each of their definitions.

You can make a pull request for the current fix, and then a new one for the recursive sequence check.

**Comment by cjmyers** [comment]
Make sure message is on bottom of the tooltip

**Comment by cjmyers** [comment]
Move orientation below role but before messages.
Combinatorial is not showing up.
If Combinatorial, then don't show sequence incomplete

Order should be:
Composite
Combinatorial
Sequence Incomplete

**Comment by cjmyers** [comment]
Sequence incomplete should not be shown when there are no child components.
